src/application/service/index.ts

```diff
@@ -101,11 +106,21 @@ export const getExistingOrCreateNewUser = new GetExistingOrCreateNewUser(
 
 export const bulkCreateUsers = new BulkCreateUsers(userRepository, roleRepository);
 
-export const exchangeAuthCode = new ExchangeAuthCode(
+const authenticationProvider = new DokobitAuthenticationProvider();
+
+export const createAuthenticationSession = new CreateAuthenticationSession(authenticationProvider);
+export const magicLinkAuthenticator = new MagicLinkAuthenticator(
   magicLinkRepository,
   generateAuthToken,
   getExistingOrCreateNewUser
 );
+export const estonianIdAuthenticator = new EstonianIdAuthenticator(
+  authenticationProvider,
+  getExistingOrCreateNewUser,
+  generateAuthToken
+);
+export const authenticatorFactory = new AuthenticatorFactory([magicLinkAuthenticator, estonianIdAuthenticator]);
```

Comment:
I guess the implementations are in the service layer because of the inversion here?